### PR TITLE
#157731613 Add Fitbit Support

### PR DESCRIPTION
**Pull request by peterwade153**
### What does this PR do?
Add Fitbit support
#### Description of Task to be completed?
This will allow user get weight and exercise information from Fitbit API.
#### How should this be manually tested?
1. Set up wger on machine, following the instructions in the ReadMe file
2. Start the app
    `python manage.py runserver`
3. Head over to your browser 
4. Register and Login
5. Select weight from the navigation bar and click on `weight overview`
6. Click `Add Fitbit`
7. If you have no fitbit account, create one, login and finally authorize wger

image

Exercise overview
image

#### What are the relevant pivotal tracker stories?
#157731613

### CODE REVIEW

File: wger/core/views/fitbit.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import base64
+import requests
+
+class Fitbit():
+    '''
+    Enables users get their weight and exercise information from fitbit
+    '''
+
+    #APP SETTINGS FOR FITBIT
+    REDIRECT_URI = 'http://127.0.0.1:8000/en/dashboard'
```
Comment (emugaya) [inline wger/core/views/fitbit.py]: These should be in the general settings file and in an environment variable.

**Review by emugaya — CHANGES_REQUESTED** [review]
Fix merge conflicts.
Stop hardcoding` urls` in your work. use environement variables and use enviroment variables to store tokens and other information

**Comment by hariclerry** [comment]
Code Looks great so far.

File: wger/core/views/fitbit.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import urllib
+import base64
+import requests
+
+class Fitbit():
+    '''
+    Enables users get their weight and exercise information from fitbit
+    '''
+
+    #APP SETTINGS FOR FITBIT
```
Comment (emugaya) [inline wger/core/views/fitbit.py]: Move these settings in the `settings file`

File: wger/core/views/fitbit.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        return self.Authorization_URI +'?'+ encoded_params
+
+    def get_access_token(self, code):
+        # gets access_token if access_code passes the code_challenge
+
+        #authentication header
+        data_str = self.CLIENT_ID +':'+ self.CLIENT_SECRET
+        auth_header = base64.b64encode(data_str.encode('utf-8'))
+        header = {
+            'Authorization' : 'Basic MjJEMjRZOmVkNjQ5OTA2N2M4MDQ5NTgyMWU1YTQxYzliODFlNWNl',
```
Comment (emugaya) [inline wger/core/views/fitbit.py]: Remove this token. Don't hardcode tokens...

File: wger/weight/views.py [hunk 4]
```diff
@@ -153,11 +155,19 @@ def overview(request, username=None):
                                      'day': max_date.day}
 
     last_weight_entries = helpers.get_last_entries(user)
+    
+    code = request.GET.get('code')
+    if code:
+        weight_data = fetch_weight_data(request, code)
+        if weight_data:
+            print(weight_data)
```
Comment (hariclerry) [inline wger/weight/views.py]: Please remember to remove this line when you are done.

File: wger/weight/static/js/weight.js [hunk 5]
```diff
@@ -60,6 +60,7 @@ $(document).ready(function () {
     if (json.length) {
       data = MG.convert.date(json, 'date');
       weightChart.data = data;
+      console.log(json)
```
Comment (hariclerry) [inline wger/weight/static/js/weight.js]: Please remember to remove this line when you are done.

**Review by hariclerry — CHANGES_REQUESTED** [review]

**Review by dondrzzy — COMMENTED** [review]
@peterwade153  Looks good

**Pull request merged**